instance/instance.go

```diff
@@ -109,6 +109,66 @@ func (s *Store) Add(w http.ResponseWriter, r *http.Request) {
 	log.Debug("add instance", log.Data{"instance": instance})
 }
 
+// UpdateDimension updates label and/or description for a specific dimension within an instance
+func (s *Store) UpdateDimension(w http.ResponseWriter, r *http.Request) {
+
+	vars := mux.Vars(r)
+	ID := vars["id"]
+	dimension := vars["dimension"]
+
+	// Get instance
+	instance, err := s.GetInstance(ID)
+	if err != nil {
+		log.Error(err, nil)
+	}
+
+	// Early return if instance is already published
+	if instance.State == "published" {
+		w.WriteHeader(http.StatusForbidden)
+		return
+	}
+
+	// Read and unmarshal request body
+	bytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Error(err, nil)
+	}
+
+	var dim *models.CodeList
+
+	err = json.Unmarshal(bytes, &dim)
+	if err != nil {
+		log.Error(err, nil)
+	}
+
+	// Update instance-dimension
+	for i := 0; i < len(instance.Dimensions); i++ {
+
+		// For the chosen dimension
+		if instance.Dimensions[i].Name == dimension {
+
+			// Assign update info, conditionals to allow updating of both or either without blanking other
+			if dim.Label != "" {
+				instance.Dimensions[i].Label = dim.Label
+			}
+			if dim.Description != "" {
+				instance.Dimensions[i].Description = dim.Description
+			}
+		}
```

Comment:
could add a break just inside the end of this if statement - don't need to continue through the loop once we've found a match as only one dimension can be named for update on this endpoint